irc_nicktrack.go

```diff
@@ -0,0 +1,103 @@
+package irc
+
+import (
+	"regexp"
+	"strings"
+)
+
+//Struct to store Channel Info
+type Channel struct {
+	Topic string
+	Mode  string
+	Users map[string]User
+}
+
+type User struct {
+	Host string
+	Mode string
+}
+
+var mode_split = regexp.MustCompile("([%@+]{0,1})(.+)") //Half-Op, //Op, //Voice
+
+func (irc *Connection) SetupNickTrack() {
+	// 353: RPL_NAMEREPLY per RFC1459
+	// will typically receive this on channel joins and when NAMES is
+	// called via GetNicksOnCHan
+	irc.AddCallback("353", func(e *Event) {
+		// get chan
+		channelName := e.Arguments[2]
+		// check if chan exists in map
+		_, ok := irc.Channels[channelName]
+
+		// if not make one
+		if ok != true {
+			irc.Channels[channelName] = Channel{Users: make(map[string]User)}
+		}
+		// split the datat into a slice
+		for _, modenick := range strings.Split(e.Message(), " ") {
+			nickandmode := mode_split.FindStringSubmatch(modenick)
+			u := User{}
+			if len(nickandmode) == 3 {
+				if nickandmode[1] == "@" {
```

Comment:
As mentioned above, these should probably be pulled from isupport, if possible, rather than hard coded.